src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -3274,21 +3278,274 @@ def _if_stmt_handler(self, node, parent):
+            clause = BinaryOperation.create(
+                BinaryOperation.Operator.EQ, Reference(type_string_symbol),
+                Literal(guard_type_repr[idx].lower(), CHARACTER_TYPE))
+
+            ifblock.addchild(clause)
+            # Add If_body
+            ifbody = Schedule(parent=ifblock)
+            self.process_nodes(parent=ifbody, nodes=stmts[idx])
+            # Replace references to the type selector variable with
+            # references to the appropriate pointer.
+            for reference in ifbody.walk(Reference):
+                symbol = reference.symbol
+                if symbol.name.lower() == selector.lower():
+                    reference.symbol = pointer_symbols[idx]
+            ifblock.addchild(ifbody)
+            currentparent = ifblock
+
+        if default_idx >= 0:
+            elsebody = Schedule(parent=currentparent)
+            currentparent.addchild(elsebody)
+            self.process_nodes(parent=elsebody, nodes=stmts[default_idx])
+
+        # Step5: Ensure that the type selector variable declaration has the
+        # pointer or a target attribute. The difficulty is that it
+        # will be in an UnknownFortranType
+        type_selector_name = selector
+        symbol_table = outer_ifblock.scope.symbol_table
+        symbol = symbol_table.lookup(type_selector_name)
+        datatype = symbol.datatype
+        # Re-create the datatype as an fparser ast
+        dummy_code = (
+            f"subroutine dummy()\n"
+            f"  {datatype.declaration}\n"
+            f"end subroutine\n")
+        parser = ParserFactory().create(std="f2008")
+        reader = FortranStringReader(dummy_code)
+        fp2_ast = parser(reader)
+        type_decl_stmt = fp2_ast.children[0].children[1].children[0]
+        attr_spec_list = type_decl_stmt.children[1]
+        attr_spec_str_list = []
+        found = False
+        if attr_spec_list:
+            for attr_spec in attr_spec_list.children:
+                attr_spec_str = attr_spec.string
+                attr_spec_str_list.append(attr_spec_str)
+                if attr_spec_str.upper() in ["TARGET", "POINTER"]:
+                    # There is already a target or pointer attribute
+                    found = True
+                    break
+        if not found:
+            # TARGET needs to be added as an attribute
+            if attr_spec_str_list:
+                # At least one attribute already exists
+                attr_spec_str_list.append("TARGET")
+                attr_spec_list = Fortran2003.Attr_Spec_List(
+                    ", ".join(attr_spec_str_list))
+            else:
+                # There are no pre-existing attributes
+                attr_spec_list = Fortran2003.Attr_Spec_List("TARGET")
+            type_decl_stmt.items = (
+                type_decl_stmt.items[0], attr_spec_list,
+                type_decl_stmt.items[2])
+            attr_spec_list.parent = type_decl_stmt
+            # pylint: disable=protected-access
```

Comment:
The fact you need this also points to potentially moving this functionality into the type class.

Reply:
A separate issue though.